std/node/_fs/_fs_dir.ts

```diff
@@ -17,25 +17,24 @@ export default class Dir {
   }
 
   read(callback?: Function): Promise<Dirent | null> {
-    return new Promise(async (resolve, reject) => {
-      try {
-        if (!this.asyncIterator) {
-          this.asyncIterator = Deno.readdir(this.path)[Symbol.asyncIterator]();
-        }
-
-        const result: Dirent | null = await (await this.asyncIterator?.next())
-          .value;
-        resolve(result ? result : null);
-
-        if (callback) {
-          callback(null, result ? result : null);
-        }
-      } catch (err) {
-        if (callback) {
-          callback(err, null);
-        }
-        reject(err);
+    return new Promise((resolve, reject) => {
+      if (!this.asyncIterator) {
+        this.asyncIterator = Deno.readdir(this.path)[Symbol.asyncIterator]();
       }
+      this.asyncIterator
+        ?.next()
```

Comment:
`?` is unnecessary here, it should rather be `!`

Reply:
thanks

Reply:
actually, it should have an `assert(this.asyncIterator)` on the line before! 😄

Reply:
we should always try to avoid not-null assertions where possible

Reply:
make sense because of `strict null-checking` 🤔 
thank you